src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -67,6 +71,21 @@ public class CISShadowMonitor implements Consumer<NetworkStateProvider.Connectio
                     .maxRetryInterval(Duration.ofMinutes(30L)).maxAttempt(Integer.MAX_VALUE)
                     .retryableExceptions(Arrays.asList(ThrottlingException.class, InternalServerException.class))
                     .build();
+    private static final RetryUtils.RetryConfig GET_CIS_SHADOW_RETRY_CONFIG =
+            RetryUtils.RetryConfig.builder()
+                    .initialRetryInterval(Duration.ofSeconds(1L))
+                    .maxRetryInterval(Duration.ofMinutes(30L))
+                    .maxAttempt(Integer.MAX_VALUE)
+                    .retryableExceptions(Collections.singletonList(Exception.class))
+                    .build();
+
+    @Getter(AccessLevel.PACKAGE)
+    @Setter(AccessLevel.PACKAGE)
+    private long mqttOperationTimeoutSeconds = 10L;
```

Comment:
get the timeout from the mqtt client. mqttClient.getOperationTimeout or something like that

Reply:
In fact, our MQTT client already has a timeout; so I'm not sure we actually need our own timeout here, it may be fine to just do a blocking `get`

Reply:
ah yeah, will adjust